homeassistant/components/alexa_local_control.py

```diff
@@ -0,0 +1,527 @@
+  If not specified, this defaults to true.
+
+- exposed_domains defines the domains that are exposed by default if
+  expose_by_default is set to true. If not specified, this defaults to the
+  following list:
+      - 'switch'
+      - 'light'
+      - 'script'
+      - 'scene'
+      - 'group'
+      - 'input_boolean'
+      - 'media_player'
+
+The following are attributes that can be applied in the customize section:
+
+- echo specifies whether or not the entity should be discoverable by Alexa. The
+  default for this attribute is controlled with the expose_by_default config
+  option.
+
+- echo_friendly_name specifies the name that Alexa will know the entity as. The
+  default for this is the entity's friendly name.
+
+Much of this code is based on work done by Bruce Locke on his ha-local-echo
+project, located at "https://github.com/blocke/ha-local-echo", originally
+released under the MIT License. The license is reproduced below:
+
+The MIT License (MIT)
+
+Copyright (c) 2016 Bruce Locke
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+import threading
+import socket
+import logging
+import json
+
+from homeassistant import util, core
+from homeassistant.const import (
+    ATTR_ENTITY_ID, SERVICE_TURN_OFF, SERVICE_TURN_ON,
+    EVENT_HOMEASSISTANT_START
+)
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_SUPPORTED_FEATURES, SUPPORT_BRIGHTNESS
+)
+
+DOMAIN = "alexa_local_control"
+
+REQUIREMENTS = ['Flask==0.11.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+# Note that the double newline at the end of
+# this string is required per the SSDP spec
+SSDP_RESPONSE = """HTTP/1.1 200 OK
+CACHE-CONTROL: max-age=60
+EXT:
+LOCATION: http://{0}:{1}/description.xml
+SERVER: FreeRTOS/6.0.5, UPnP/1.0, IpBridge/0.1
+ST: urn:schemas-upnp-org:device:basic:1
+USN: uuid:Socket-1_0-221438K0100073::urn:schemas-upnp-org:device:basic:1
+
+"""
+
+DESCRIPTION_XML_RESPONSE = """<?xml version="1.0" encoding="UTF-8" ?>

```

Comment:
Let's move this to line 282 since it's not a constant that other components would need to get at.